homeassistant/components/buienradar/sensor.py

```diff
@@ -225,12 +212,14 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         timeframe,
     )
 
-    dev = []
-    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
-        dev.append(BrSensor(sensor_type, config.get(CONF_NAME), coordinates))
-    async_add_entities(dev)
+    entities = [
+        BrSensor(sensor_type, config.get(CONF_NAME, "Buienradar"), coordinates)
+        for sensor_type in SENSOR_TYPES
+    ]
+
+    async_add_entities(entities)
 
-    data = BrData(hass, coordinates, timeframe, dev)
+    data = BrData(hass, coordinates, timeframe, entities)
```

Comment:
A future improvement would be to either use the data update coordinator or the dispatch helper to signal the entities to update instead of accessing the entities from outside. The latter is not a good pattern.

Reply:
I agree, there is more room for improvement.